client/orderbook/orderbook.go

```diff
@@ -604,3 +613,30 @@ func (ob *OrderBook) BestFill(sell bool, qty uint64) ([]*Fill, bool) {
 func (ob *OrderBook) BestFillMarketBuy(qty, lotSize uint64) ([]*Fill, bool) {
 	return ob.sells.bestFill(qty, true, lotSize)
 }
+
+func (ob *OrderBook) SetMatchesSummary(matches map[uint64]uint64, ts uint64) {
+	ob.matchSummaryMtx.Lock()
+	defer ob.matchSummaryMtx.Unlock()
+	newMatchesSummary := make([]*MatchSummary, len(matches))
+	i := 0
+	for rate, qty := range matches {
+		newMatchesSummary[i] = &MatchSummary{
+			Rate: rate,
+			Qty:  qty,
+			Age:  ts,
+		}
+		i++
+	}
+	if ob.matchesSummary == nil {
+		ob.matchesSummary = newMatchesSummary
+		return
+	}
+	ob.matchesSummary = append(newMatchesSummary, ob.matchesSummary...)
```

Comment:
This grows forever. Maybe a good idea to prune sometimes? Does not look very big, but not sure whats best. Maybe is len is more than 100_000 chop off the first 10_000?

Reply:
Now if matchesSummary is bigger than 100, it will slice out the first ones added to the queue

Reply:
I still think you don't need to chop off so much. I suggest 
```
if len(ob.matchesSummary) > maxLength {
        ob.matchesSummary = ob.matchesSummary[len(ob.matchesSummary)-maxLength+100:]
}
```
So, if the thing is over 1000 set it to 900.